torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -363,8 +383,15 @@ def init_unsharded_param(self):
                 self._global_size,
                 self._global_stride,
             )
-        self._unsharded_param = nn.Parameter(unsharded_param)
-        self._unsharded_param.requires_grad_(self.sharded_param.requires_grad)
+        if self._unsharded_param is not None:
+            assert torch._dynamo.compiled_autograd.compiled_autograd_enabled
+            with torch.no_grad():
+                alloc_storage(self._unsharded_param)
+                self._unsharded_param.copy_(unsharded_param)
+        else:
+            self._unsharded_param = nn.Parameter(unsharded_param, requires_grad=self.sharded_param.requires_grad)
+        if torch._dynamo.compiled_autograd.compiled_autograd_enabled:
+            self.all_gather_outputs = []
```

Comment:
Note to self, this needs to be moved to after free_storage, see https://github.com/pytorch/pytorch/pull/128105/files